src/Umbraco.Web.UI.Client/src/packages/documents/documents/property-dataset-context/document-property-dataset.context.ts

```diff
@@ -1,9 +1,68 @@
-import type { UmbDocumentDetailModel, UmbDocumentVariantModel } from '../types.js';
+import type { UmbDocumentDetailModel, UmbDocumentVariantModel, UmbDocumentWorkspaceContext } from '../types.js';
+import { UMB_DOCUMENT_CONFIGURATION_CONTEXT } from '../global-contexts/index.js';
 import { UmbContentPropertyDatasetContext } from '@umbraco-cms/backoffice/content';
 import type { UmbDocumentTypeDetailModel } from '@umbraco-cms/backoffice/document-type';
+import type { UmbVariantId } from '@umbraco-cms/backoffice/variant';
+import type { DocumentConfigurationResponseModel } from '@umbraco-cms/backoffice/external/backend-api';
+import type { UmbControllerHost } from '@umbraco-cms/backoffice/controller-api';
+import { observeMultiple } from '@umbraco-cms/backoffice/observable-api';
+import type { UmbVariantPropertyGuardRule } from '@umbraco-cms/backoffice/property';
 
 export class UmbDocumentPropertyDatasetContext extends UmbContentPropertyDatasetContext<
 	UmbDocumentDetailModel,
 	UmbDocumentTypeDetailModel,
 	UmbDocumentVariantModel
-> {}
+> {
+	#dataSetVariantId?: UmbVariantId;
+	#documentConfiguration?: DocumentConfigurationResponseModel;
+	#allowEditInvariantFromNonDefault: boolean | undefined = undefined;
+
+	constructor(host: UmbControllerHost, dataOwner: UmbDocumentWorkspaceContext, variantId?: UmbVariantId) {
+		super(host, dataOwner, variantId);
+
+		this.#dataSetVariantId = variantId;
+
+		this.consumeContext(UMB_DOCUMENT_CONFIGURATION_CONTEXT, async (context) => {
+			this.#documentConfiguration = (await context?.getDocumentConfiguration()) ?? undefined;
+			this.#allowEditInvariantFromNonDefault = this.#documentConfiguration?.allowEditInvariantFromNonDefault;
+			this.#enforceAllowEditInvariantFromNonDefault();
```

Comment:
Wouldn't it make sense only to run the `#enforceAllowEditInvariantFromNonDefault` when this.#allowEditInvariantFromNonDefault is set to false?

Reply:
I have updated the code 👍